File: linear_search/linear_search.cpp
#include <bits/stdc++.h>

using namespace std;

//parameters are linear_search(array address, size of array, key to find)
bool linear_search(int arr[], int size , int key) {

	for (int i = 0 ; i < size; i++) {
		if (key == arr[i])return true;
	}

	return false;
}

int main() {

	//example
	int arr[] = {2, 3, 4, 5, 7, 8};

	//linear_search function returns true if element is found and false if not found
	cout << linear_search(arr, 6, 3) << '\n';
	cout << linear_search(arr, 6, 1) << '\n';

	return 0;
}

File: linear_search/linear_search.go
package main

import "fmt"

func linear_search (size int, key int, arr []int) string{
	
	for i := 0; i < size; i++ {
    	if arr[i]==key{
			return "found"
		}
	}
	
	return "not found"

}

func main() {

	arr := []int{1, 2, 3, 4, 5}
	ans := linear_search(5,2,arr)
	fmt.Println(ans)
	
}
